cdap-api/src/main/java/co/cask/cdap/api/data/DatasetContext.java

```diff
@@ -59,4 +57,43 @@
   <T extends Dataset> T getDataset(String name, Map<String, String> arguments)
     throws DatasetInstantiationException;
 
+  /**
+   * Calling this means that the dataset is not used by the caller any more,
+   * and the DatasetContext is free to dismiss or reuse it. The dataset must
+   * have been acquired through this context using {@link #getDataset(String)}
+   * or {@link #getDataset(String, Map)}. It is up to the implementation of the
+   * context whether this dataset is discarded, or whether it is reused as the
+   * result for subsequent {@link #getDataset} calls with the same arguments.
+   * This may be influenced by resource constraints, expiration policy, or
+   * advanced configuration.
+   * <p>
+   * Note: In the current implementation, this is equivalent to {@link #discardDataset(Dataset)}.
```

Comment:
This was copied from the old DatasetProvider. I think it is useful to mention that. Where else would you mention it? Since the implementations are not part of API, it would not show up to users.